wire/msgtx.go

```diff
@@ -338,6 +432,22 @@ func (msg *MsgTx) BtcEncode(w io.Writer, pver uint32) error {
 		return err
 	}
 
+	// If the pver is set to WitnessVersion, and the Flags field for the
+	// MsgTx aren't 0x00, then this indicates the transaction is to be
+	// encoded using the new witness inclusionary structure defined in
+	// BIP0141.
+	if pver == WitnessVersion {
+		// After the txn's Version field, we include two additional
+		// bytes specific to the witness encoding. The first byte is an
+		// always 0 marker byte, which allows decoders to distinguish a
+		// serialized transaction with witnesses from a regular (legacy)
+		// one. The second byte is the Flag field, which at the moment is
+		// always 0x01, but way be extended in the future to accommodate
+		// auxiliary non-commited fields.
+		w.Write([]byte{0x00})
```

Comment:
Also, I'd do a single write with a `constant` that has []byte{0x00, 0x01} along with the appropriate commenting to explain it.